modules/dnn/src/layers/layers_common.cpp

```diff
@@ -55,128 +55,144 @@ std::string makeName(const std::string& str1, const std::string& str2)
 {
     return str1 + str2;
 }
-
 bool getParameter(const LayerParams &params, const std::string& nameBase, const std::string& nameAll,
-                  int &parameterH, int &parameterW, bool hasDefault = false, const int& defaultValue = 0)
+                  std::vector<size_t>& parameter, bool hasDefault = false, const int& defaultValue = 0)
 {
     std::string nameH = makeName(nameBase, std::string("_h"));
     std::string nameW = makeName(nameBase, std::string("_w"));
     std::string nameAll_ = nameAll;
-    if(nameAll_ == "")
-    {
+    if (nameAll_ == "")
         nameAll_ = nameBase;
-    }
 
     if (params.has(nameH) && params.has(nameW))
     {
-        parameterH = params.get<int>(nameH);
-        parameterW = params.get<int>(nameW);
+        CV_Assert(params.get<int>(nameH) >= 0 && params.get<int>(nameW) >= 0);
+        parameter.push_back(params.get<int>(nameH));
+        parameter.push_back(params.get<int>(nameW));
         return true;
     }
     else
     {
         if (params.has(nameAll_))
         {
             DictValue param = params.get(nameAll_);
-            parameterH = param.get<int>(0);
-            if (param.size() == 1)
-            {
-                parameterW = parameterH;
-            }
-            else if (param.size() == 2)
-            {
-                parameterW = param.get<int>(1);
+            for (int i = 0; i < param.size(); i++) {
+                CV_Assert(param.get<int>(i) >= 0);
+                parameter.push_back(param.get<int>(i));
             }
-            else
-            {
-                return false;
+            if (parameter.size() == 1) {
+                parameter.push_back(parameter.back());
             }
             return true;
         }
         else
         {
-            if(hasDefault)
+            if (hasDefault)
             {
-                parameterH = parameterW = defaultValue;
+                parameter.push_back(defaultValue);
+                parameter.push_back(defaultValue);
                 return true;
             }
-            else
-            {
                 return false;
-            }
```

Comment:
Please return `else` or at least align the line because now it looks like
```cpp
if (hasDefault)
{
    parameter.push_back(defaultValue);
    parameter.push_back(defaultValue);
    return true;
}
    return false;
```

Reply:
done